docs/testing/playwright/02-e2e-testing.md

```diff
@@ -0,0 +1,125 @@
+
+```html title="my-component.e2e.html"
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf8" />
+
+    <!-- Replace with the path to your entrypoint -->
+    <script src="./build/test-app.esm.js" type="module"></script>
+    <script src="./build/test-app.js" nomodule></script>
+  </head>
+  <body>
+    <my-component first="Stencil"></my-component>
+  </body>
+</html>
+```
+
+```ts title="my-component.e2e.ts"
+import { expect } from '@playwright/test';
+import { test } from '@stencil/playwright';
+
+test.describe('my-component', () => {
+  test('should render the correct name', async ({ page }) => {
+    // The path here is the path to the www output relative to the dev server root directory
+    await page.goto('/my-component/my-component.e2e.html');
+
+    // Rest of test
+  });
+});
+```
+
+#### `page.setContent()`
+
+The `setContent()` method allows tests to define their own HTML code on a test-by-test basis. This pattern is helpful if the HTML for a test is small, or to
+avoid affecting other tests is using the `page.goto()` pattern and modifying a shared HTML template file. With this pattern, the `script` tags pointing to Stencil
+entry code will be automatically injected into the generated HTML.
+
+```ts title="my-component.e2e.ts"
+import { expect } from '@playwright/test';
+import { test } from '@stencil/playwright';
+
+test.describe('my-component', () => {
+  test('should render the correct name', async ({ page }) => {
+    await page.setContent('<my-component first="Stencil"></my-component>');
+
+    // Rest of test
+  });
+});
+```
+
+#### `page.waitForChanges()`
+
+The `waitForChanges()` method is a utility for waiting for Stencil components to rehydrate after an operation that results in a re-render:
+
+```ts title="my-component.e2e.ts"
+import { expect } from '@playwright/test';
+import { test } from '@stencil/playwright';
+
+test.describe('my-component', () => {
+  test('should render the correct name', async ({ page }) => {
+    // Assume we have a template setup with the `my-component` component and a `button`
+    await page.goto('/my-component/my-component.e2e.html');
+
+    const button = page.locator('button');
+    // Assmume clicking the button changes the `@Prop()` values on `my-component`
+    button.click();
+    await page.waitForChanges();
+  });
+});
+```
+
+## Running Tests
+
+By default, the adapter will execute all tests in a project with a `.e2e.ts` file suffix. This can be modified by passing the
+[`testDir`](https://playwright.dev/docs/api/class-testproject#test-project-test-dir) and/or [`testMatch`](https://playwright.dev/docs/api/class-testproject#test-project-test-match)
+configuration options as overrides to `createStencilPlaywrightConfig()`.
+
+To run tests, either run `npx playwright test` from the **root** of the Stencil project, or update the project's `test` script in the `package.json` file to run the
+Playwright command.
```

Comment:
What do you think about switching the order of these two sections? I don't want to imply that `npx stencil test` will run these playwright tests, and am thinking it'd make sense to say "run `npx playwright test` right out of the gate:

```suggestion
To run tests, either run `npx playwright test` from the **root** of the Stencil project, or update the project's `test` script in the `package.json` file to run the
Playwright command.

By default, the adapter will execute all tests in a project with a `.e2e.ts` file suffix. This can be modified by passing the
[`testDir`](https://playwright.dev/docs/api/class-testproject#test-project-test-dir) and/or [`testMatch`](https://playwright.dev/docs/api/class-testproject#test-project-test-match)
configuration options as overrides to `createStencilPlaywrightConfig()`.
```